hydradx/model/amm/omnipool_amm.py

```diff
@@ -84,65 +146,122 @@ def swap_lrna(
         old_state: dict,
         old_agents: dict,
         trader_id: string,
-        delta_R: float,
-        delta_Q: float,
-        i: int
+        delta_Ra: float,
+        delta_Qa: float,
+        i: int,
+        fee_assets: float = 0,
+        fee_lrna: float = 0
 ) -> tuple:
     """Compute new state after LRNA swap"""
 
     new_state = copy.deepcopy(old_state)
     new_agents = copy.deepcopy(old_agents)
 
-    if delta_Q == 0 and delta_R != 0:
-        delta_Q = swap_lrna_delta_Qi(old_state, delta_R, i)
-    elif delta_R == 0 and delta_Q != 0:
-        delta_R = swap_lrna_delta_Ri(old_state, delta_Q, i)
+    if delta_Qa < 0:
+        delta_Q = -delta_Qa
+        delta_R = old_state['R'][i] * -delta_Q / (delta_Q + old_state['Q'][i]) * (1 - fee_assets)
+        delta_L = -delta_Q * (1 + (1 - fee_assets) * old_state['Q'][i] / (old_state['Q'][i] + delta_Q))
+        delta_Ra = -delta_R
+    elif delta_Ra > 0:
```

Comment:
Actually I guess it would be old_state['R'][i] * (1 - fee_assets)
I also notice there is no check to make sure the trader actually has that much LRNA

Reply:
yep I believe your formula is correct, with the fee